test/format/ops.c

```diff
@@ -1845,8 +1845,15 @@ col_insert_resolve(TABLE *table, void *arg)
     do {
         WT_ORDERED_READ(max_rows, table->rows_current);
         for (i = 0, p = cip->insert_list; i < WT_ELEMENTS(cip->insert_list); ++i, ++p) {
-            if (*p == max_rows + 1) {
-                testutil_assert(__wt_atomic_casv32(&table->rows_current, max_rows, max_rows + 1));
+            /*
+             * A thread may have allocated a record number that is now less than or equal to the
+             * current maximum number of rows. In this case, simply reset the insert list.
+             * Otherwise, update the maximum number of rows with the newly inserted record.
+             */
+            if (*p > 0 && *p <= max_rows + 1) {
+                if (*p == max_rows + 1)
+                    testutil_assert(
+                      __wt_atomic_casv32(&table->rows_current, max_rows, max_rows + 1));
```

Comment:
If you're confident that the issue is a race condition between threads inserting records, then I think this is a good approach. 

From reading the ticket, it seems you were tossing up between a race condition and bulk insert incorrectly updating this field after failing to load all records. Did you end up proving this one way or another?

I'm also curious about why this only popped up after mirroring - I don't think this code changed, right?

Reply:
> From reading the ticket, it seems you were tossing up between a race condition and bulk insert incorrectly updating this field after failing to load all records. Did you end up proving this one way or another?

With the reproducer, the bulk insert code that I thought could be an issue (setting the _wrong_ number of rows) was never hit.

> I'm also curious about why this only popped up after mirroring - I don't think this code changed, right?

Yes, I agree, I cannot figure out how it is related to the mirror branch. I checked and the table impacted is not a mirrored one.

Reply:
> With the reproducer, the bulk insert code that I thought could be an issue (setting the wrong number of rows) was never hit.

OK, thanks.

> Yes, I agree, I cannot figure out how it is related to the mirror branch. I checked and the table impacted is not a mirrored one.

Fair enough. When you say on the ticket that it only started hitting on the merge of WT-8372, did you try your repro before/after? I would be a lot more comfortable if we could show this was just a coincidence.

I appreciate I'm probably being difficult - if Sue doesn't think it's necessary, I'm happy to defer to her :-)

Reply:
> I appreciate I'm probably being difficult - if Sue doesn't think it's necessary, I'm happy to defer to her :-)

I think it's a good idea, let me verify it. I will get back to you once I have done the testing.

Reply:
I have added a couple of comments to the ticket that I think explains the current behaviour.

Reply:
Wow, nice work - thanks for digging on on that.